corehq/apps/locations/fixtures.py

```diff
@@ -70,47 +70,87 @@ def __call__(self, user, version, last_sync=None, app=None):
-                    location_id__in=location_ids
-                )
-
-        location_db = _location_footprint(locations)
-
-        if not should_sync_locations(last_sync, location_db, user):
-            return []
 
-        root = Element('fixture',
-                       {'id': self.id,
-                        'user_id': user.user_id})
+        all_locations = _all_locations(user)
+
+        if not should_sync_locations(last_sync, all_locations, user):
+            return []
 
+        root_node = Element('fixture', {'id': self.id, 'user_id': user.user_id})
         root_locations = filter(
-            lambda loc: loc.parent is None, location_db.by_id.values()
+            lambda loc: loc.parent is None, all_locations.by_id.values()
         )
 
         if root_locations:
-            _append_children(root, location_db, root_locations)
-        return [root]
+            _append_children(root_node, all_locations, root_locations)
+        return [root_node]
 
 
 location_fixture_generator = LocationFixtureProvider()
 
 
+def _all_locations(user):
+    if toggles.SYNC_ALL_LOCATIONS.enabled(user.domain):
+        return LocationSet(SQLLocation.active_objects.filter(domain=user.domain))
+    else:
+        leaf_locations = {leaf_location for leaf_location in _gather_leaf_locations(user)}
+        return _location_footprint(leaf_locations)
+
+
+def _gather_leaf_locations(user):
+    """Returns all the leaf locations for which we want ancestors
+    """
+    # From the root most location we want (expand_from), we traverse down the
+    # tree until we get all the leaf-most locations we want. We populate the
+    # fixture with all ancestors of these desired leaves later
+
+    user_location = user.sql_location
+    user_locations = set([user_location]) if user_location is not None else set()
+    user_locations = user_locations | {location for location in _gather_multiple_locations(user)}
+
+    if not user_locations:
+        raise StopIteration()
+
+    for user_location in user_locations:
+        location_type = user_location.location_type
+        expand_from = location_type.expand_from or location_type
+        expand_to = location_type.expand_to
+        root = _get_root(user.domain, user_location, expand_from)
+
+        for root_location in root:
+            for leaf in _get_leaves(root_location, expand_to):
+                yield leaf
+
+
+def _gather_multiple_locations(user):
+    """If the project has multiple locations enabled, returns all the extra
+    locations the user is assigned to.
+    """
+    if user.project.supports_multiple_locations_per_user:
+        location_ids = [loc.location_id for loc in user.locations]
+        for location in SQLLocation.active_objects.filter(location_id__in=location_ids):
+            yield location
+
+
+def _get_root(domain, user_location, expand_from):
+    """From the users current location, returns the highest location they want in their fixture
```

Comment:
I don't think this is the highest location they want in their fixture - it's where the expansion starts.  The highest location is always going to be one or more top-level locations.